### ISSUE TITLE
 How can I dynamically translate a route?

### ISSUE THREAD

**Issue by PaoloDiaz27** [Greentube/localize-router#12]
I try to translate a route dynamically in a component, I need to do that because I navigate from a component to other by a button, but before of navigate I need to do something else.
I do this
 
this.localize.translateRoute('restaurantList').subscribe((path: string) => {
            this.router.navigate(['restaurantList']);
          });

I don't know if I do something wrong, I just downloaded localize-router, everything works fine, but I cann't translate the route in the component. Can you help me please. Thanks!

**Comment by PaoloDiaz27** [comment]
When i initialize "private localize: LocalizeRouterService" in the constructor, the app don't load and show this error in the console. Maybe you know what happen?

![errorlocalizerouter1](https://cloud.githubusercontent.com/assets/24846410/21572410/972c6c68-cea6-11e6-8f3f-9d96574b6545.png)

**Comment by meeroslav** [comment]
Hi @PaoloDiaz27 ,

Sorry for waiting. 
For your first question, answer would be to use translated `path` variable instead of `restaurantList` string in `router.navigate` method:

    this.localize.translateRoute('restaurantList').subscribe((path: string) => {
        this.router.navigate([path]);
    });

As for the second question, I can only see from your log that `resetConfig` fails because your route config is not valid. Can you provide a link to your repo or example of your routes?

**Comment by PaoloDiaz27** [comment]
These are my routes

// Route Configuration
export const routes: Routes = [
    {path: '', redirectTo: '/index', pathMatch: 'full'},
    {path: 'index', component: IndexComponent},
    {path: 'another', component: AnotherComponent},
    {path: 'dashboard', component: DashboardComponent},
    {path: 'contactUs', component: ContactUsComponent},
    {path: 'restaurantList', component: RestaurantListComponent},
    {path: 'detailRestaurant', component: DetailRestaurantComponent},
    {path: 'faq', component: FaqComponent}
];

and in the app.module.ts I call them that

RouterModule.forRoot(routes),
LocalizeRouterModule.forRoot(routes)

The problem is when I initialize "LocalizeRouterService" in the constructor from component, the app stop working, if I comment this variable the app works again. I don't know what is the problem then

**Comment by meeroslav** [comment]
Is there a possibility you can create a small repo where this issue can be reproduced?

**Comment by PaoloDiaz27** [comment]
Hi @meeroslav  

I want to tell you that I have solved the problem. In the app.module.ts I was calling a service that did nothing, then deleted that service and initialized "LocalizeRouterService" in the constructor, the app continued to run normal.
Then, I use the following statement

this.localize.translateRoute('restaurantList').subscribe((path: string) => {
    this.router.navigate([path]);
});

But it does not worked, in another issue, I saw another statement and with it works.

this.localize.translateRoute('/restaurantList', true).subscribe( path => this.router.navigate([path]) );

I want to thank you for the help and congratulation you for the module, it works very well and It's what I was looking for
